assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -385,6 +385,31 @@ describe( 'createSettingsStore store', () => {
 			} );
 		} );
 
+		test.each( [
+			[ 'haveSettingsChanged' ],
+			[ '__dangerousHaveSettingsChanged' ],
+		] )( 'should have %s selector', ( selector ) => {
+			const selectors = storeDefinition.selectors;
+			expect( typeof selectors[ selector ] ).toBe( 'function' );
+		} );
+
+		describe.each( [
+			[ 'haveSettingsChanged' ],
+			[ '__dangerousHaveSettingsChanged' ],
+		] )( '%s', ( selector ) => {
+			it( 'should use provided validateHaveSettingsChanged function', () => {
+				const validateHaveSettingsChanged = jest.fn();
+				storeDefinition = createSettingsStore( ...STORE_ARGS, {
+					settingSlugs: [ 'isSkyBlue' ],
+					validateHaveSettingsChanged,
+					registry,
+				} );
+
+				storeDefinition.selectors[ selector ]();
+				expect( validateHaveSettingsChanged ).toHaveBeenCalled();
+			} );
+		} );
+
```

Comment:
Let's add tests to ensure the "safe" selector (`haveSettingsChanged`) never throws an exception when `__dangerousHaveSettingsChanged` _would_ throw.